eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/http/SendMessageRequestProtocolResolver.java

```diff
@@ -42,98 +42,59 @@ public static CloudEvent buildEvent(Header header, Body body) throws ProtocolHan
-                        cloudEventBuilder = cloudEventBuilder.withExtension(entry.getKey(), entry.getValue());
-                    }
-                }
-                event = cloudEventBuilder.build();
-            } else if (StringUtils.equals(SpecVersion.V03.toString(), protocolVersion)) {
-                cloudEventBuilder = CloudEventBuilder.v03();
-                cloudEventBuilder = cloudEventBuilder.withId(sendMessageRequestBody.getBizSeqNo())
-                    .withSubject(sendMessageRequestBody.getTopic())
-                    .withType("eventmeshmessage")
-                    .withSource(URI.create("/"))
-                    .withData(content.getBytes(StandardCharsets.UTF_8))
-                    .withExtension(ProtocolKey.REQUEST_CODE, code)
-                    .withExtension(ProtocolKey.ClientInstanceKey.ENV, env)
-                    .withExtension(ProtocolKey.ClientInstanceKey.IDC, idc)
-                    .withExtension(ProtocolKey.ClientInstanceKey.IP, ip)
-                    .withExtension(ProtocolKey.ClientInstanceKey.PID, pid)
-                    .withExtension(ProtocolKey.ClientInstanceKey.SYS, sys)
-                    .withExtension(ProtocolKey.ClientInstanceKey.USERNAME, username)
-                    .withExtension(ProtocolKey.ClientInstanceKey.PASSWD, passwd)
-                    .withExtension(ProtocolKey.VERSION, version.getVersion())
-                    .withExtension(ProtocolKey.LANGUAGE, language)
-                    .withExtension(ProtocolKey.PROTOCOL_TYPE, protocolType)
-                    .withExtension(ProtocolKey.PROTOCOL_DESC, protocolDesc)
-                    .withExtension(ProtocolKey.PROTOCOL_VERSION, protocolVersion)
-                    .withExtension(SendMessageRequestBody.BIZSEQNO, sendMessageRequestBody.getBizSeqNo())
-                    .withExtension(SendMessageRequestBody.UNIQUEID, sendMessageRequestBody.getUniqueId())
-                    .withExtension(SendMessageRequestBody.PRODUCERGROUP,
-                        sendMessageRequestBody.getProducerGroup())
-                    .withExtension(SendMessageRequestBody.TTL, sendMessageRequestBody.getTtl());
-                if (StringUtils.isNotEmpty(sendMessageRequestBody.getTag())) {
-                    cloudEventBuilder = cloudEventBuilder.withExtension(SendMessageRequestBody.TAG, sendMessageRequestBody.getTag());
-                }
-                if (sendMessageRequestBody.getExtFields() != null && sendMessageRequestBody.getExtFields().size() > 0) {
-                    for (Map.Entry<String, String> entry : sendMessageRequestBody.getExtFields().entrySet()) {
-                        cloudEventBuilder = cloudEventBuilder.withExtension(entry.getKey(), entry.getValue());
-                    }
-                }
-                event = cloudEventBuilder.build();
+            switch (SpecVersion.parse(sendMessageRequestHeader.getProtocolVersion())) {
+                case SpecVersion.V1:
```

Comment:
enum switch case label must be the unqualified name of an enumeration constant.
```
case V1:
         cloudEventBuilder = CloudEventBuilder.v1();
         break;
 case V03:
         cloudEventBuilder = CloudEventBuilder.v03();
         break;
```

Reply:
> enum switch case label must be the unqualified name of an enumeration constant.
> 
> ```
> case V1:
>          cloudEventBuilder = CloudEventBuilder.v1();
>          break;
>  case V03:
>          cloudEventBuilder = CloudEventBuilder.v03();
>          break;
> ```
right but that would mean we would be doing a static import of SpecVersion right?

Reply:
This class is just an enumeration class, which is already used in the original code.

Reply:
thank you for the help, I had some confusion around the syntax. I could use some help setting up my local. Having build issues because of which I am not able to use a lot of IDE stuffs.

Reply:
Enumeration types are defined in switch statements. For each case clause, the compiler simply checks for the presence of text in the enumeration, and if it is allowed to specify fully qualified class names in the case clause, then the compiler must perform a useless step.